app/scripts/lib/createDupeReqFilterStream.ts

```diff
@@ -42,24 +42,23 @@ const makeExpirySet = () => {
 };
 
 /**
- * Returns a middleware that filters out requests whose ids we've already seen.
+ * Returns a "through" stream that filters out requests whose ids we've already seen.
  * Ignores JSON-RPC notifications, i.e. requests with an `undefined` id.
  *
- * @returns The middleware function.
+ * @returns The stream object.
  */
-export default function createDupeReqFilterMiddleware(): JsonRpcMiddleware<
-  unknown,
-  void
-> {
+export default function createDupeReqFilterStream() {
   const seenRequestIds = makeExpirySet();
-  return function filterDuplicateRequestMiddleware(req, _res, next, end) {
-    if (req.id === undefined) {
-      // JSON-RPC notifications have no ids; our only recourse is to let them through.
-      return next();
-    } else if (!seenRequestIds.add(req.id)) {
-      log.info(`RPC request with id "${req.id}" already seen.`);
-      return end();
+  return createThroughStream((chunk, _, cb) => {
```

Comment:
This is just a wrapper for `Transform` in `readable-stream` already. Can we just use that directly?

https://github.com/rvagg/through2/blob/ebe3d0b736115ffca785336301119979df70a708/through2.js#L70

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/24672/commits/a8a25a0cc29058e114445f03bece395c53b52e00